.github/workflows/ruby_tests.yml

```diff
@@ -4,94 +4,26 @@ on:
 env:
   RAILS_ENV: test
   DATABASE_URL: postgresql://postgres:@localhost/test
   DATABASE_CLEANER_ALLOW_REMOTE_DATABASE_URL: true
+
 jobs:
   rubocop:
-    runs-on: ubuntu-latest
-    env:
-      BUNDLE_FOREMAN: '0'
-      BUNDLE_WITHOUT: development:test
-    steps:
-      - uses: actions/checkout@v2
-      - name: Setup Ruby
-        uses: ruby/setup-ruby@v1
-        with:
-          ruby-version: 2.7
-          bundler-cache: true
-      - name: Run rubocop
-        run: bundle exec rubocop -P --format github
-  test_ruby:
-    runs-on: ubuntu-latest
+    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
+    with:
+      command: bundle exec rubocop -P --format github
+      
+  test:
+    name: Ruby
     needs: rubocop
-    env:
-      BUNDLE_WITHOUT: journald:development:console:libvirt
-    services:
-      postgres:
-        image: postgres:12.1
-        ports: ['5432:5432']
-        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
-    strategy:
-      fail-fast: true
-      matrix:
-        foreman-repo: [theforeman/foreman]
-        foreman-core-branch: [develop]
-        ruby-version: [2.7]
-        node-version: [12]
-    steps:
-      - run: |
-          sudo apt-get update
-          sudo apt-get install build-essential libcurl4-openssl-dev zlib1g-dev libpq-dev
-      - name: Install chromedriver
-        uses: nanasess/setup-chromedriver@master
-      - uses: actions/checkout@v2
-        with:
-          repository: ${{ matrix.foreman-repo }}
-          ref: ${{ matrix.foreman-core-branch }}
-      - uses: actions/checkout@v2
-        with:
-          path: foreman_puppet
-      - name: Setup Bundler
-        run: |
-          echo "gem 'foreman_puppet', path: './foreman_puppet'" > bundler.d/foreman_puppet.local.rb
-      - name: Setup Ruby
-        uses: ruby/setup-ruby@v1
-        with:
-          ruby-version: ${{ matrix.ruby-version }}
-          bundler-cache: true
-      - name: Setup Node
-        if: github.event_name != 'push'
-        uses: actions/setup-node@v1
-        with:
-          node-version:  ${{ matrix.node-version }}
-      - name: Prepare test DB
-        if: github.event_name != 'push'
-        run: |
-          bundle exec rake db:create
-          bundle exec rake db:migrate
-      - name: Restore NPMs
-        uses: actions/cache@v2
-        with:
-          path: node_modules
-          key: ${{ runner.os }}-modules-${{ matrix.node-version }}-${{ hashFiles('package.json') }}
-      - name: Prepare webpack
-        if: github.event_name != 'push'
-        run: |
-          npm install
-          bundle exec rake webpack:compile
-      - name: Run plugin tests
-        if: github.event_name != 'push'
-        run: |
-          bundle exec rake test:foreman_puppet
-      - name: Run access permission test
-        if: ${{ github.event_name != 'push' }}
-        run: |
-          bundle exec rake test TEST="test/unit/foreman/access_permissions_test.rb"
```

Comment:
This steps has not yet executed in the reusable action.

Reply:
Related discussion: https://github.com/theforeman/actions/pull/16